### ISSUE TITLE
Surge 1.9.0 - Gig Performer - Loosing selected preset

### ISSUE THREAD

**Issue by macsaif** [surge-synthesizer/surge#4723]
**Bug Description:**
After updating from 1.6.x to 1.9.0 I am having a problem with recalling the gig in Gig Performer 3. After saving the gig file and restarting the Gig Performer I have different presets selected in Surge in all rackspaces as before closing the Gig Performer. This problem is only with Surge. Other VST plugins work well.

**Surge Version**
1.9.0

* Version:
*VST3, 64bit

**Reproduction Steps:**
2. Click on '...'
3. Scroll down to '...'

**Expected Behavior:**

**Screenshots:**
If applicable, add screenshots/GIF/videos to help explain your problem.

**Computer Information (please complete the following information):**
 - OS: Windows 10
 - Host: Gig Performer
 - Version: 3.8.1

**Additional Information:**
If I save the gig file with Save as legacy format it seems that I can reload the saved file correctly - but this is not an usual way to save the gig files.

**Comment by mkruselj** [comment]
Hello!

To me this potentially sounds like a Gig Performer corner case issue related to saving plugin data chunk, considering any other host we've had people using so far didn't have this sort of problem so far...

Does the gig file only contain Surge? If so, would you like to share it (ZIP it up first)?

**Comment by baconpaul** [comment]
Another thought. Was your 16 the vst2 surge and 19 is vst3? Could that be confusing your setup somehow?

**Comment by macsaif** [comment]
The gig filé contains approx 50 rackspaces with lot of Instruments (pianoteq, vb3, ni, etc....). Maybe thé VST2/VST3 can be the problém. Now I am using VST3(no other possibility) . In previous version maybe I háve used VST2 but I can nőt check it because the 1.6 is replaced by 1.9

**Comment by macsaif** [comment]
I have tried to  replace all surge plugins in my gig by deleting the existing one and inserting a new one but the problem is the same. Than I have searched whether is there any similar problem and found that the Gig performer had the same problem with Kontakt when the predictive loading was selected. When I am deselecting the predictive loading then the saving and recalling works correctly, however in that case I have higher CPU load because all rackspaces are loaded.
Link to my discussion in Gig Performer:
https://community.gigperformer.com/t/surge-vst-loosing-settings/7315/12

**Comment by mkruselj** [comment]
What happens if you create an entirely new .gig from scratch, with 1.9 VST3? Does it recall correctly?

**Comment by DavidJameson** [comment]
I'm one of the GP Developers --- happy to give you ( mkruselj ) a free NFR so you can do further testing. Please reach out to me via email through our website. There are no issues with Kontakt and predictive loading (there might have been in the long distant pass but I've confirmed with people who use that feature all the time that this works)

Basically, all GP does is asks the plugin to return its state, we save that state  and then when a gig is reloaded, we send that state back to the plugin. We don't modify or otherwise interpret that state in any way.

**Comment by baconpaul** [comment]
Right; @mkruselj and I were thinking that OP may have been using the VST2 of 1.6 (which we stopped distributing after 1.6) and the swap with the 1.9 upgrade from VST2 to VST3 may have broken the rack?

Does GP run on mac? If so I'd find time to debug in an NFR! Thanks for the kind offer.

**Comment by DavidJameson** [comment]
Yes, GP is both Mac and Windows. Please use the contact form on our website so I can get your email address and I'll set you up.

**Comment by baconpaul** [comment]
Great done. Thanks

**Comment by baconpaul** [comment]
So thanks for the nfr! Gig performer is cool

Anyway I made a patch with surge 19 and it saved and restored without problem. So I think the problem here is making one with the vst2 from 16 and trying to load it into the vst3 which is 19 - but some more info on what exactly the issue is would be great!
